File: Util/jrand/t/t.m
/* * Objective-C Augments - A small, miscellaneous set of Objective-C String and Data
 * augmentations
 * Copyright (C) 2011- nicerobot
 *
 * This file is part of Objective-C Augments.
 *
 * Objective-C Augments is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Objective-C Augments is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Objective-C Augments.  If not, see <http://www.gnu.org/licenses/>.
 */


#import <Foundation/Foundation.h>
#import "JRandom.h"

int main (int argc, const char * argv[]) {
  NSAutoreleasePool * pool = [[NSAutoreleasePool alloc] init];
  int seed = 0;
  switch (argc) {
    case 1:
      seed = 0;
      break;
    default:
      seed = atoi(argv[1]);
      break;
  }
  JRandom *r = [[JRandom alloc] initWithSeed:seed];
  for (int i=0; i<10; i++) {
    printf("%08X\n",[r nextInt]);
  }
  [pool drain];
  return 0;
}

File: Util/jrand/t/t.java
/*
 * C/Objective-C Utilities - A small, miscellaneous set of C/Objective-C utilities
 * Copyright (C) 2011- nicerobot
 *
 * This file is part of C/Objective-C Utilities.
 *
 * C/Objective-C Utilities is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * C/Objective-C Utilities is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with C/Objective-C Utilities.  If not, see <http://www.gnu.org/licenses/>.
 */

import java.util.Random;

class t {
  public static void main(String... args) {
    int seed = 0;
    if (args.length > 0) {
      seed = Integer.parseInt(args[0]);
    }
    Random r = new Random(seed);
    for (int i=0; i<10; i++) {
      System.out.printf("%08X\n",r.nextInt());
    }    
  }
}
